File: src/main/java/com/mpe85/grampa/input/impl/StringBufferInputBuffer.java
package com.mpe85.grampa.input.impl;

import com.mpe85.grampa.input.InputBuffer;

/**
 * An {@link InputBuffer} implementation using a {@link StringBuffer}.
 * 
 * @author mpe85
 *
 */
public class StringBufferInputBuffer extends CharSequenceInputBuffer {
	
	public StringBufferInputBuffer(final StringBuffer stringBuffer) {
		super(stringBuffer);
		this.stringBuffer = stringBuffer;
	}
	
	@Override
	public int getCodePoint(final int index) {
		return stringBuffer.codePointAt(index);
	}
	
	
	private final StringBuffer stringBuffer;
	
}

File: src/main/kotlin/com/mpe85/grampa/input/impl/StringBufferInputBuffer.kt
package com.mpe85.grampa.input.impl

import com.mpe85.grampa.input.InputBuffer

/**
 * An [InputBuffer] implementation using a [StringBuffer].
 *
 * @author mpe85
 * @property[stringBuffer] A string buffer backing the input buffer
 */
public class StringBufferInputBuffer(private val stringBuffer: StringBuffer) : CharSequenceInputBuffer(stringBuffer) {

    override fun getCodePoint(index: Int): Int = stringBuffer.codePointAt(index)
}
